app/src/main/java/com/tks/trendsmeme/presentation/MainActivity.kt

```diff
@@ -24,35 +39,230 @@ import androidx.compose.runtime.remember
             }
         }
     }
 }
 
+@Composable
+fun EnterMobileNumberPage() {
+
+    Column(modifier = Modifier
+        .fillMaxSize()
+        .background(color = Color.Black),
+        verticalArrangement = Arrangement.Center, horizontalAlignment = Alignment.CenterHorizontally) {
+
+        AddAppLogo()
+        AddMobileEditText()
+        AddContinueButton()
+        SeparatorWithOrCircle()
+        SocialMediaLogin()
+        AddPrivacyText()
+    }
+
+
+
+
+
+}
+
+@Composable
+fun AddContinueButton() {
+
+    val context=LocalContext.current
+    val modifier = Modifier
+        .fillMaxWidth()
+        .height(80.dp)
+        .padding(16.dp)
+    CustomComponents().RoundedCornerButton(
+        onClick = { sendOtp(context) },
+        text = "Continue",
+        modifier
+    )
+}
+
+fun sendOtp(context: Context) {
+    "Otp sent to your mobile".showToast(context,Toast.LENGTH_LONG)
+}
+
+@Composable
+fun AddAppLogo() {
+    Image(painter = painterResource(id = R.drawable.app_logo_slogan), contentDescription = "app icon",Modifier.size(150.dp),Alignment.TopStart)
+
+}
+
+//@Composable
+//fun AddMobileEditText(){
+//    var mobileNumber by remember { mutableStateOf("") }
+//    val modifier = Modifier
+//        .fillMaxWidth()
+//        .height(100.dp)
+//        .padding(16.dp)
+//    CustomComponents().ReusableEditText(
+//        modifier = modifier,
+//        value = mobileNumber,
+//        onValueChange = { mobileNumber = it },
+//        label = "Enter Mobile Number",
+//        keyboardType = KeyboardType.Phone)
+//}
+@Composable
+fun AddMobileEditText() {
+    var mobileNumber by remember {
+        mutableStateOf(
+            ""
+        )
+    }
+    val modifier = Modifier
+        .fillMaxWidth()
+        .padding(16.dp)
+
+    Box(modifier = Modifier
+        .wrapContentSize()
+        .clip(CircleShape)
+        .padding(16.dp)
+        .background(colorResource(id = R.color.edittext_back_color))) {
+        OutlinedTextField(
+            value = mobileNumber,
+            onValueChange = { mobileNumber = it },
+            textStyle = TextStyle(color= Color.White),
+            leadingIcon = {
+                Row(verticalAlignment =Alignment.CenterVertically, modifier = Modifier.padding(10.dp)) {
+                    Image(
+                        painter = painterResource(id = R.drawable.baseline_flag_24),
+                        contentDescription = "India Flag",
+                        modifier = Modifier.size(24.dp)
+                    )
+                    Spacer(
+                        modifier =Modifier.width(
+                            8.dp
+                        )
+                    )
+                    Text("+91",color=Color.White)
+                }
+            },
+            label = { Text("Enter Mobile Number", color = Color.White) },
+            modifier = modifier,
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Phone)
+        )
+    }
```

Comment:
**Suggestion:** Refactor the `AddMobileEditText` composable to separate the UI composition from the business logic, enhancing the clarity and reusability of the code. [maintainability, importance: 7]
```suggestion
@Composable
fun AddMobileEditText(mobileNumber: String, onMobileNumberChange: (String) -> Unit) {
    val modifier = Modifier
        .fillMaxWidth()
        .padding(16.dp)

    Box(modifier = Modifier
        .wrapContentSize()
        .clip(CircleShape)
        .padding(16.dp)
        .background(colorResource(id = R.color.edittext_back_color))) {
        OutlinedTextField(
            value = mobileNumber,
            onValueChange = onMobileNumberChange,
            textStyle = TextStyle(color= Color.White),
            leadingIcon = {
                Row(verticalAlignment =Alignment.CenterVertically, modifier = Modifier.padding(10.dp)) {
                    Image(
                        painter = painterResource(id = R.drawable.baseline_flag_24),
                        contentDescription = "India Flag",
                        modifier = Modifier.size(24.dp)
                    )
                    Spacer(
                        modifier =Modifier.width(
                            8.dp
                        )
                    )
                    Text("+91",color=Color.White)
                }
            },
            label = { Text("Enter Mobile Number", color = Color.White) },
            modifier = modifier,
            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Phone)
        )
    }
}
```